packages/api/src/models/medical/facility.ts

```diff
@@ -14,13 +29,56 @@ export class FacilityModel extends BaseModel<FacilityModel> implements Facility
         cxId: {
           type: DataTypes.UUID,
         },
+        oid: {
+          type: DataTypes.STRING,
+          defaultValue: "",
+        },
+        facilityNumber: {
+          type: DataTypes.INTEGER,
+          defaultValue: 0,
+        },
         data: {
           type: DataTypes.JSONB,
         },
+        cqOBOActive: {
+          type: DataTypes.BOOLEAN,
+          defaultValue: false,
+        },
+        cwOBOActive: {
+          type: DataTypes.BOOLEAN,
+          defaultValue: false,
+        },
+        cqOBOOID: {
+          type: DataTypes.STRING,
+          allowNull: true,
+        },
+        cwOBOOID: {
+          type: DataTypes.STRING,
+          allowNull: true,
+        },
+        type: {
+          type: DataTypes.ENUM(...Object.values(FacilityType)),
+          defaultValue: FacilityType.initiatorAndResponder,
+        },
       },
       {
         ...BaseModel.modelOptions(sequelize),
         tableName: FacilityModel.NAME,
+        hooks: {
+          async beforeCreate(attributes) {
+            const curMaxNumber = (await FacilityModel.max("facilityNumber", {
+              where: {
+                cxId: attributes.cxId,
+              },
+            })) as number;
+            const org = await getOrganizationOrFail({ cxId: attributes.cxId });
+            const facNumber = curMaxNumber ? curMaxNumber + 1 : OID_ID_START;
+            attributes.oid = `${Config.getSystemRootOID()}.${OIDNode.organizations}.${
+              org.organizationNumber
+            }.${OIDNode.locations}.${facNumber}`;
+            attributes.facilityNumber = facNumber;
+          },
+        },
```

Comment:
Could try using `executeOnDBTx()` so we can lock the facility's rows when creating a new one. I'd just query the org before that DB tx to minimize lock time.